rust/flexbuffers/src/flexbuffer.rs

```diff
@@ -0,0 +1,89 @@
+use std::ops::{RangeBounds, Rem, Bound};
+use std::slice::SliceIndex;
+use std::fmt::Debug;
+
+use crate::reader::ReadLE;
+
+pub trait FlexBuffer: Debug + PartialEq + IntoIterator + AsRef<[u8]> + Default + Clone {
```

Comment:
Can we reduce the set of traits? 

* AsRef<[u8]> is definitely needed 
*  PartialEq, Debug, and IntoIterator, I'm not so sure
* I think Clone should instead be Copy, since accessing a flexbuffer field can return a new Reader and should be a cheap operation.

Reply:
So under the hood `Clone` should be using `Copy` in cases of a `&[u8]`. Custom buffers on the other hand (like the tokio/bytes impl) have more complicated data structures and can't implement Copy. 

That being said, it's still generally a very lightweight operation. Some buffers just need to be able to handle reference counting logic (which can't be done in a Copy trait, since the Copy impl is either automatic / derived or not possible).